geoapps/utils/models.py

```diff
@@ -0,0 +1,192 @@
+        )
+
+        for attr, item in kwargs.items():
+            try:
+                setattr(self, attr, item)
+            except AttributeError:
+                continue
+
+    @property
+    def center(self):
+        """Prism center"""
+        return self._center
+
+    @center.setter
+    def center(self, value):
+        self._center = value
+        self._vertices = None
+
+    @property
+    def length(self):
+        """"""
+        return self._length
+
+    @length.setter
+    def length(self, value):
+        self._length = value
+        self._vertices = None
+
+    @property
+    def width(self):
+        """"""
+        return self._width
+
+    @width.setter
+    def width(self, value):
+        self._width = value
+        self._vertices = None
+
+    @property
+    def depth(self):
+        """"""
+        return self._depth
+
+    @depth.setter
+    def depth(self, value):
+        self._depth = value
+        self._vertices = None
+
+    @property
+    def dip(self):
+        """"""
+        return self._dip
+
+    @dip.setter
+    def dip(self, value):
+        self._dip = value
+        self._vertices = None
+
+    @property
+    def azimuth(self):
+        """"""
+        return self._azimuth
+
+    @azimuth.setter
+    def azimuth(self, value):
+        self._azimuth = value
+        self._vertices = None
+
+    @property
+    def reference(self):
+        """Point of rotation to be 'center' or 'top'"""
+        return self._reference
+
+    @reference.setter
+    def reference(self, value):
+        if value not in ["center", "top"]:
+            raise ValueError("'reference' point should be 'center' or 'top'.")
+        self._reference = value
+        self._vertices = None
+
+    @property
+    def vertices(self):
+        """
+        Prism eight corners in 3D space
+        """
+
+        if getattr(self, "_vertices", None) is None:
+            x1, x2 = [
+                -self.length / 2.0 + self.center[0],
+                self.length / 2.0 + self.center[0],
+            ]
+            y1, y2 = [
+                -self.width / 2.0 + self.center[1],
+                self.width / 2.0 + self.center[1],
+            ]
+            z1, z2 = [
+                -self.depth / 2.0 + self.center[2],
+                self.depth / 2.0 + self.center[2],
+            ]
+
+            block_xyz = np.asarray(
+                [
+                    [x1, x2, x1, x2, x1, x2, x1, x2],
+                    [y1, y1, y2, y2, y1, y1, y2, y2],
+                    [z1, z1, z1, z1, z2, z2, z2, z2],
+                ]
+            )
+
+            xyz = rotate_vertices(block_xyz.T, self.center, self.dip, self.azimuth)
+
+            if self.reference == "top":
+                offset = np.mean(xyz[4:, :], axis=0) - self._center
+                self._center = self._center - offset
+                xyz -= offset
+
+            self._vertices = xyz
+
+        return self._vertices
+
+
+def rotate_vertices(xyz, center, phi, theta):
```

Comment:
The input variable names are a bit confusing. Maybe rename so it takes in dip and azimuth, since it converts them to phi and theta in the function.